packages/react/src/transactions/adapters/PublicationCallGateway.ts

```diff
@@ -237,7 +325,21 @@ export class PublicationCallGateway implements ICreatePostCallGateway {
     };
   }
 
-  private async uploadPublicationMetadata(request: CreatePostRequest): Promise<string> {
+  private async resolveCreateCommentRequestArg(
+    request: CreateCommentRequest,
+  ): Promise<CreatePublicCommentRequestArg> {
+    return {
+      profileId: request.profileId,
+      publicationId: request.publicationId,
+      contentURI: await this.uploadPublicationMetadata(request),
+      collectModule: resolveCollectModule(request),
+      referenceModule: resolveReferenceModule(request),
+    };
+  }
+
+  private async uploadPublicationMetadata(
+    request: CreatePostRequest | CreateCommentRequest,
+  ): Promise<string> {
     return this.upload(createPublicationMetadata(request));
```

Comment:
Should we model some known error when the upload failed? 
Given that `upload` is externally provided by the user it may throw some exceptions that would be good to forward back to the `useCreatePost/Comment` hook in place of `error` so the user may decide to handle those in a way we can't foresee.
Any thoughts?

Reply:
I've done some work here: https://github.com/lens-protocol/lens-sdk/pull/61

Worth taking into account my other comment: https://github.com/lens-protocol/lens-sdk/pull/57#discussion_r1060600649

There is an opportunity to reduce the complexity of this code. Maybe with a generic or an abstract class?